src/tyro/extras/app/_tyro_app.py

```diff
@@ -0,0 +1,48 @@
+from __future__ import annotations
+
+from typing import Any, Callable, Dict, TypeVar
+
+import tyro
+
+CallableT = TypeVar("CallableT", bound=Callable)
+
+
+_subcommands: Dict[str, Callable] | None = None
```

Comment:
I would not use a global dict for it. Because you may have different CLIs in one project.
So i prefer my solution that needs to make a instance by the user: 

https://github.com/jedie/cli-base-utilities/blob/31ebd3812571dd8068d2dbff9151af5ab6d810da/cli_base/tyro_commands.py#L27